scalalib/src/mill/scalalib/TestModuleUtil.scala

```diff
@@ -173,15 +209,217 @@ private[scalalib] object TestModuleUtil {
+        Some(result)
+      } else {
+        None
+      }
+    }
+
+    val groupFolderData = filteredClassLists match {
+      case Nil => Seq((Task.dest, prepareTestSelectorFolder(Nil, Task.dest), 0, ""))
+      case Seq(singleTestClassList) =>
+        Seq((
+          Task.dest,
+          prepareTestSelectorFolder(singleTestClassList, Task.dest),
+          singleTestClassList.length,
+          ""
+        ))
+      case multipleTestClassLists =>
+        val maxLength = multipleTestClassLists.length.toString.length
+        multipleTestClassLists.zipWithIndex.map { case (testClassList, i) =>
+          val groupPromptMessage = testClassList match {
+            case Seq(single) => single
+            case multiple =>
+              TestModuleUtil.collapseTestClassNames(
+                multiple
+              ).mkString(", ") + s", ${multiple.length} suites"
+          }
+          val paddedIndex = mill.internal.Util.leftPad(i.toString, maxLength, '0')
+          val folderName = testClassList match {
+            case Seq(single) => single
+            case multiple =>
+              s"group-$paddedIndex-${multiple.head}"
+          }
+
+          (
+            Task.dest / folderName,
+            prepareTestSelectorFolder(testClassList, Task.dest / folderName),
+            testClassList.length,
+            groupPromptMessage
+          )
+        }
+    }
+
+    val jobs = Task.ctx() match {
+      case j: Ctx.Jobs => j.jobs
+      case _ => 1
+    }
+
+    val maxProcessLength = jobs.toString.length
+    val groupLength = groupFolderData.length
+    val maxGroupLength = groupLength.toString.length
+
+    val subprocessFutures = for {
+      processIndex <- 0 until jobs
+      groupIndex <- 0 until groupLength
+      (baseFolder, selectorFolder, numTests, groupPromptMessage) = groupFolderData(groupIndex)
+      if (processIndex == 0) || (processIndex < numTests)
+    } yield {
+      val paddedGroupIndex = mill.internal.Util.leftPad(groupIndex.toString, maxGroupLength, '0')
+      val paddedProcessIndex =
+        mill.internal.Util.leftPad(processIndex.toString, maxProcessLength, '0')
+      val processFolder = baseFolder / processIndex.toString
+
+      Task.fork.async(
+        processFolder,
+        s"${paddedGroupIndex}-${paddedProcessIndex}",
+        groupPromptMessage
+      ) { logger =>
+        // force run when processIndex == 0 (first subprocess), even if there are no tests to run
+        // to force the process to go through the test framework setup/teardown logic
+        s"${baseFolder.last}" -> runTestRunnerSubprocess(
+          groupIndex,
+          processIndex,
+          processFolder,
+          selectorFolder,
+          force = processIndex == 0,
+          logger
+        )
+      }
+    }
+
+    val logScheduler = new java.util.concurrent.ScheduledThreadPoolExecutor(1)
+
+    logScheduler.scheduleWithFixedDelay(
+      new Runnable {
+        override def run(): Unit = {
```

Comment:
I think this can be shortened using the lambda syntax `() => {...}` without explicitly instantiating an anonymous subclass of `Runnable`